cns/restserver/restserver.go

```diff
@@ -379,6 +383,115 @@ func (service *HTTPRestService) deleteNetwork(w http.ResponseWriter, r *http.Req
 	log.Response(service.Name, resp, resp.ReturnCode, ReturnCodeToString(resp.ReturnCode), err)
 }
 
+// Handles CreateHnsNetwork requests.
+func (service *HTTPRestService) createHnsNetwork(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[Azure CNS] createHnsNetwork")
+
+	var err error
+	returnCode := 0
+	returnMessage := ""
+
+	var req cns.CreateHnsNetworkRequest
+	err = service.Listener.Decode(w, r, &req)
+	log.Request(service.Name, &req, err)
+
+	if err != nil {
+		returnMessage = fmt.Sprintf("[Azure CNS] Error. Unable to decode input request.")
+		returnCode = InvalidParameter
+	} else {
+		switch r.Method {
```

Comment:
could we combine create and delete and check for method to be POST or DELETE here?

Reply:
We are not following that style in the other CNS APIs. I would prefer to keep it that way.